spring-statemachine-core/src/main/java/org/springframework/statemachine/service/DefaultStateMachineService.java

```diff
@@ -128,6 +128,18 @@ public void releaseStateMachine(String machineId, boolean stop) {
 		}
 	}
 
+	/**
+	 * Determines if the given machine identifier denotes a known managed state machine.
+	 *
+	 * @param machineId machine identifier
+	 * @return true if machineId denotes a known managed state machine currently in memory
+	 */
+	public boolean isStateMachineManaged(String machineId) {
```

Comment:
Maybe this could just be `hasStateMachine`. Should we think about adding this to an interface as well?

Reply:
Ok, thanks. I'll rename that method when I merge. Think I'll add that to an interface with `2.2.x` as theoretically we should not do that kind of changes between patch versions(In case someone went ahead and created custom StateMachineService impl).